scala/scala.bzl

```diff
@@ -257,54 +255,67 @@ scala_repl = rule(
     toolchains = ['@io_bazel_rules_scala//scala:toolchain_type'],
 )
 
-def scala_repositories(maven_servers = ["http://central.maven.org/maven2"]):
-  _scala_maven_import_external(
-      name = "io_bazel_rules_scala_scala_library",
-      artifact = "org.scala-lang:scala-library:2.11.12",
-      jar_sha256 =
-      "0b3d6fd42958ee98715ba2ec5fe221f4ca1e694d7c981b0ae0cd68e97baf6dce",
-      licenses = ["notice"],
-      server_urls = maven_servers,
-  )
-  _scala_maven_import_external(
-      name = "io_bazel_rules_scala_scala_compiler",
-      artifact = "org.scala-lang:scala-compiler:2.11.12",
-      jar_sha256 =
-      "3e892546b72ab547cb77de4d840bcfd05c853e73390fed7370a8f19acb0735a0",
-      licenses = ["notice"],
-      server_urls = maven_servers,
-  )
+def scala_repositories(
+    scala_version_shas = (_default_scala_version(),
+                          _default_scala_version_jar_shas()),
+    maven_servers = ["http://central.maven.org/maven2"]):
+  (scala_version, scala_version_jar_shas) = scala_version_shas
+  major_version = _extract_major_version(scala_version)
+
+  _new_scala_default_repository(
+      scala_version = scala_version,
+      scala_version_jar_shas = scala_version_jar_shas,
+      maven_servers = maven_servers)
+
+  scala_extra_jar_shas = {
```

Comment:
shouldn't this also be a parameter like the `_default_scala_version_jar_shas`? I might be missing something here

Reply:
As these only need to be specified per major scala version I decided to not to have them as parameters. This means that we'll have to update them to support other major scala versions.

I'd rather leave it like this for now and do another PR to add that functionality